UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,193 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_TRANSFORM_HPP
+#define UNICORN_VIDEO_TRANSFORM_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+
+#define GLM_FORCE_RADIANS
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+/**
+ * @brief Base class for every object that can be translated or rotated
+ */
+class Transform
+{
+public:
+    /** @brief Default constructor */
+    Transform();
+
+    /** @brief Default destructor */
+    virtual ~Transform() = default;
+
+    /**
+     * @brief Sets the direction of the view
+     *
+     * @param[in] direction new direction vectorr
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction);
+
+    /**
+     * @brief Sets the direction of the view
+     *
+     * @param[in] direction new direction vector
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction, glm::vec3 upVector);
+
+    /**
+     * @brief Sets up vector
+     *
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void SetUp(glm::vec3 upVector);
+
+    /**
+     * @brief      Sets the world coordinates
+     *
+     * @param[in]  x world axis
+     * @param[in]  y world axis
+     * @param[in]  z world axis
+     */
+    UNICORN_EXPORT void SetWorldAxes(glm::vec3 x, glm::vec3 y, glm::vec3 z);
+
+    /**
+     * @brief Sets translate
+     *
+     * @param[in]  translate new translate vector
+     */
+    UNICORN_EXPORT void SetTranslation(glm::vec3 translate);
+
+    /**
+     * @brief Returns direction
+     *
+     * @return direction
+     */
+    UNICORN_EXPORT glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Return right vector
+     *
+     * @return right vector
+     */
+    UNICORN_EXPORT glm::vec3 GetRight() const;
+
+    /**
+     * @brief Returns up vector
+     *
+     * @return up vector
+     */
+    UNICORN_EXPORT glm::vec3 GetUp() const;
+
+    /**
+     * @brief Return translate
+     *
+     * @return translate
+     */
+    UNICORN_EXPORT glm::vec3 GetTranslation() const;
+
+    /**
+     * @brief Returns model matrix
+     *
+     * @attention Don't forget to recalculate @sa Calculate before
+     *
+     * @return model matrix
+     */
+    UNICORN_EXPORT glm::mat4 const& GetModelMatrix() const;
+
+    /**
+     * @brief Scales object
+     *
+     * @param[in] scale basis scale factors
+     *
+     * Lineary scales each basis component of the mesh vectors by the given scale factors
```

Comment:
`Linearly`

Also move the description _before_ `@param`
Check https://github.com/Godlike/Unicorn/wiki/Coding-style-guideline#class-comments